pkg/addonmanager/controllers/agentdeploy/utils.go

```diff
@@ -517,9 +518,45 @@ func getManifestConfigOption(agentAddon agent.AgentAddon,
 		}
 	}
 
+	emc := agentAddon.GetAgentAddonOptions().ManifestConfigs
+	for _, mc := range emc {
+		index := containsResourceIdentifier(manifestConfigs, mc.ResourceIdentifier)
+		if index == -1 {
+			manifestConfigs = append(manifestConfigs, mc)
+			continue
+		}
+
+		for _, rule := range mc.FeedbackRules {
+			if !containsFeedbackRule(manifestConfigs[index].FeedbackRules, rule) {
+				manifestConfigs[index].FeedbackRules = append(manifestConfigs[index].FeedbackRules, rule)
+			}
+		}
+
+		if mc.UpdateStrategy != nil {
+			manifestConfigs[index].UpdateStrategy = mc.UpdateStrategy
+		}
+	}
 	return manifestConfigs
 }
 
+func containsResourceIdentifier(mcs []workapiv1.ManifestConfigOption, ri workapiv1.ResourceIdentifier) int {
+	for index, mc := range mcs {
+		if reflect.DeepEqual(mc.ResourceIdentifier, ri) {
+			return index
+		}
+	}
+	return -1
+}
+
+func containsFeedbackRule(rules []workapiv1.FeedbackRule, rule workapiv1.FeedbackRule) bool {
+	for _, r := range rules {
+		if reflect.DeepEqual(r, rule) {
```

Comment:
actually we may only compare the name of the rule. It won't work if you have two rules with the same name.

Reply:
you mean the jsonpath.name? so if they have the same name but a different path, should we override the old one with the agentOption.ManifestConfigs?

Reply:
Yes, we have to decide override or not, but comparing name only should be good enough.